types/coin.go

```diff
@@ -200,8 +200,15 @@ func (coins Coins) IsValid() bool {
 		}
 
 		lowDenom := coins[0].Denom
+		seenDenoms := make(map[string]bool)
+		seenDenoms[strings.ToUpper(lowDenom)] = true
+
 		for _, coin := range coins[1:] {
-			if strings.ToLower(coin.Denom) != coin.Denom {
+			denomUpper := strings.ToUpper(coin.Denom)
```

Comment:
I just want to check one thing (this may be right).

We compare coins as the upper case variant, assuming we are case-insensitive here: `atom` and `ATOM` are considered duplicated.

However, we sort them based on whatever the *unmodified* input denom was. Thus `eth` < `ATOM`, which is a bit odd.

Reply:
removed the `ToUpper` everywhere for consistency